go/libraries/doltcore/doltdb/durable/index.go

```diff
@@ -584,3 +588,22 @@ func (is doltDevIndexSet) RenameIndex(ctx context.Context, oldName, newName stri
 
 	return doltDevIndexSet{is.vrw, is.ns, am}, nil
 }
+
+// searchForCaseInsensitiveIndexName searches through the index names in this index set looking for a case-insensitive
+// match against |targetName|. If found, the address is returned, along with the exact case name. If no match was
+// found, a nil address is returned, along with an empty string.
+func (is doltDevIndexSet) searchForCaseInsensitiveIndexName(ctx context.Context, targetName string) (foundAddr *hash.Hash, foundName string, err error) {
```

Comment:
Think we usually return the zero-hash instead of nil for this kind of thing

Reply:
Check with hash.IsEmpty()

Reply:
yeah... I have a thing against zero-objects – `nil` just seems so much more obvious to me that the value is invalid. I agree that it's better to be consistent though, so I'll change to `hash.IsEmpty()` instead of a nil check.